python/gpusim_server.py

```diff
@@ -95,17 +119,20 @@ def do_POST(self):
             return
         self.do_json_POST(smiles, ids, scores)
 
-    def deserialize_results(self, output_qba):
-            data_reader = QtCore.QDataStream(output_qba)
-            return_count = data_reader.readInt()
-            smiles, ids, scores = [], [], []
-            for i in range(return_count):
-                smiles.append(data_reader.readString().decode("utf-8"))
-            for i in range(return_count):
-                ids.append(data_reader.readString().decode("utf-8"))
-            for i in range(return_count):
-                scores.append(data_reader.readFloat())
-            return return_count, smiles, ids, scores
+    def deserialize_results(self, request_num, output_qba):
+        data_reader = QtCore.QDataStream(output_qba)
+        returned_request = data_reader.readInt()
+        if request_num != returned_request:
+            raise RuntimeError("Incorrect result ID returned!")
+        return_count = data_reader.readInt()
+        smiles, ids, scores = [], [], []
```

Comment:
eh, you can do this all in one loop. list comprehensions also don't seem inappropriate.

Reply:
How can I do it all in 1 loop?  The pipe exports in order:
111112222233333
reading all in 1 loop with 1 read after another would be trying to read:
123123123123123

Which wouldn't match how the backend exports.

Reply:
oh, duh, I see how it's a pipe. shame you didn't make a serializable results object :/